### DISCUSSION TITLE
hakoniwa-unity-drone-model、Unity Editorでのエラー

### DISCUSSION THREAD

**Discussion by Keiichi-Ota** [toppers/hakoniwa#204] [Q&A]
![hakoniwa](https://github.com/user-attachments/assets/11263abf-46d3-4034-9cdd-6fa08246e742)

箱庭でドローンが動かせることに興味があり、最近、インストールを始めた初心者です。
一通りマニュアルを読み、ひとまず、インストールをしました。
Unityのアプリ、model.exeの方は問題なく動かせまして、ジョイスティックで動かすところまでできました。
しかしながら、Unity EditorでPLAYすると、添付のエラーが出ており、先へ進めませんでした。
"Can not create pdu channel!! channel_id=0"
"channel_id=2"のエラーが出る場合もあります。
Unity Editorのバージョンを色々変更しましたが、症状に変化がありませんでした。
Genarateするcore_config、custom、pdu_configなどのファイルも、model.exeと同じフォルダにあるものと比較したりしたのですが、大きな違いはなさそうでした。

解決方法のヒントになるアドバイスを頂けると助かります。

**Comment by buildko89** [comment]
@Keiichi-Ota さん
初めまして、Windows側でメンテやっている牧野です。

手順の確認させてください。実際、インストール行った際に、参照したマニュアルってのは、どれになりますか？
報告頂いた内容見る限り、自分でUnity側をビルドしているようですが、現状、v2.8.0のリリース中ですので、px4側とのversionが
合わないかもしれません。
動作確認するのであれば、以下のインストーラにてインストールすれば動作確認はできます。

https://github.com/toppers/hakoniwa-px4sim/releases/download/v2.7.0/hakoniwa-px4-win_v2.7.0.zip

もしくは、unityで作成したDroneAppWinのcustom.jsonを各batファイルで指定することで回避できないでしょうか？

**Reply by Keiichi-Ota** [reply]
早速のご連絡ありがとうございます。
バージョンの違いの不整合がどこかにあるようですね。

①参照したマニュアル類
マニュアルは、以下を参照しました。

１）箱庭ドローンシミュレータ 準備編-Windows環境の利用での事前準備
[hakoniwa-px4sim/docs/manual/windows_preinstall.md at main · toppers/hakoniwa-px4sim · GitHub](https://github.com/toppers/hakoniwa-px4sim/blob/main/docs/manual/windows_preinstall.md)

２）箱庭ドローンシミュレータ Windows版インストーラ利用方法-インストーラ利用とサンプルアプリ動作確認
[hakoniwa-px4sim/docs/manual/windows_hakowin_installer.md at main · toppers/hakoniwa-px4sim · GitHub](https://github.com/toppers/hakoniwa-px4sim/blob/main/docs/manual/windows_hakowin_installer.md)
 
３）箱庭ドローンシミュレータ 利用編-Windows環境の利用セットアップと動作検証
[hakoniwa-px4sim/docs/manual/windows_install&operation_PX4環境_WSL2版.md at main · toppers/hakoniwa-px4sim · GitHub](https://github.com/toppers/hakoniwa-px4sim/blob/main/docs/manual/windows_install%26operation_PX4%E7%92%B0%E5%A2%83_WSL2%E7%89%88.md#143-qgc%E3%81%AE%E8%B5%B7%E5%8B%95)

②インストールのバージョン
インストールの際は、v2.7.0を使っています。
gitより、”hakoniwa-px4-win_v2.7.0.zip”をダウンロードしました。
おそらくご案内頂いたものと同じかと思います。

③その他
「custom.jsonを各batファイルで指定する」という方法が分からずでした。

**Reply by Keiichi-Ota** [reply]
「custom.jsonを各batファイルで指定する」は、run_***,bat内のcustom.jsonを指定している部分について、UnityのGenerateで作成したものを指定する、ということですね。
その設定をしましたところ、Unity Editorからの起動に成功しました。
Unity上にて、「http 1.1 404 not found」のエラーは出ておりますが、STARTボタンが押せ、動かせました。
ご助言ありがとうございました。

となりますと、結果としまして、私の場合、どこでインストールや設定を間違えたことになりますでしょうか。
もし今後のバージョンアップに際し、留意した方が良い点がありましたら、ご教授ください。
![hakoniwa2](https://github.com/user-attachments/assets/c244fb91-5fda-4a08-8094-04c105f2c354)
![hakoniwa3](https://github.com/user-attachments/assets/1daf3bb9-9abf-4809-942f-f7ee3331f5f5)

**Reply by buildko89** [reply]
custom.jsonの件、分かり難くてすみませんでした。今回のv2.8.0で、custom.jsonの内容が大幅に変わっており、インストール手順などは変更予定です。少々お待ちください。

httpのErrorですが、以下の対応で治らないでしょうか？

https://github.com/toppers/hakoniwa-document/tree/main/troubleshooting/unity#unity%E8%B5%B7%E5%8B%95%E6%99%82%E3%81%ABnewtonsoftjson%E3%81%8C%E3%81%AA%E3%81%84%E3%81%A8%E3%81%84%E3%81%86%E3%82%A8%E3%83%A9%E3%83%BC%E3%81%8C%E5%87%BA%E3%82%8B

https://github.com/toppers/hakoniwa-document/blob/main/troubleshooting/unity/README.md#grpc-%E3%81%AE%E3%83%A9%E3%82%A4%E3%83%96%E3%83%A9%E3%83%AA%E5%88%A9%E7%94%A8%E7%AE%87%E6%89%80%E3%81%8C%E3%82%A8%E3%83%A9%E3%83%BC%E5%87%BA%E5%8A%9B%E3%81%97%E3%81%A6%E3%81%84%E3%82%8B

**Reply by tmori** [reply]
HTTP エラーは、音源がないから怒られているだけで、実害はないですね。

**Reply by Keiichi-Ota** [reply]
音源を設定したところ、HTTPのエラーは消えました。
音は出るようになりました。
ありがとうございました。
